### PR TITLE
staging-next: another iteration

### PR DESCRIPTION
**Pull request by vcunat**
This might be **the last** iteration before forking 19.03, so let me /cc @lheckemann.

### CODE REVIEW

**Comment by vcunat** [comment]
@FRidh: something seems wrong about python packages and their build inputs.  Many of them now fail to find things &ndash; in $PATH or includes or just dependency detection.  Most of [new failures](https://hydra.nixos.org/eval/1506333?filter=x86_64-linux&compare=1506218&full=) seem to be like that.

**Comment by FRidh** [comment]
@vcunat this is because I set `strictDeps = true;` in `buildPythonPackage`. Many were fixed but not all. The remainder can be fixed on `master`.

**Comment by vcunat** [comment]
I see, it should certainly help cross-compilation and probably clean up overall over longer term.  I'm not sure this period immediately before forking off stable is a good moment, as probably many fixes will need to be cherry-picked.

File: pkgs/development/python-modules/fiona/default.nix [hunk 1]
```diff
@@ -14,24 +15,35 @@ buildPythonPackage rec {
 
   CXXFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-std=c++11";
 
+  nativeBuildInputs = [
+    gdal # for gdal-config
```
Comment (dotlambda) [inline pkgs/development/python-modules/fiona/default.nix]: @FRidh Just wanted to let you know that I believe this is a situation where a package needs to be in `nativeBuildInputs` as well as `buildInputs`.

Comment (FRidh) [reply pkgs/development/python-modules/fiona/default.nix]: Probably yes. `gdal-config` requires it to be in `nativeBuildInputs`, but we also build against it so also in `buildInputs`.

**Comment by FRidh** [comment]
Python failure of `jupyter`/`nbconvert` is transient.

**Comment by lheckemann** [comment]
This won't magically be ready in the next hour, will it? :/ I guess we'll have to rebase it into the release branch after the fact. :(

**Comment by FRidh** [comment]
I think this can go in. Aarch64 is currently broken, but that can be fixed up.

**Comment by FRidh** [comment]
@vcunat @lheckemann how to proceed to get this into `staging-19.03`?

**Comment by FRidh** [comment]
I've merged staging-next into both master and staging-19.03.

**Pull request merged**